### PR TITLE
Add dashboards rpm assemble support in opensearch-build repo

### LINKED ISSUE #1545: [RPM M1] Add a new block to call the generation code for RPM
Tasks | Estimate | Status | Notes | Dependencies
-- | -- | -- | -- | --
The generation code should pull the artifacts from the build workflow to a temporary location | 1 | Complete |   | Build workflow must provide usable artifacts
The code will call existing install function to install plugins on min artifacts | 1 | Complete |   |  
After installation, the code will execute a tool or utility to wrap all the content into a RPM package | 5 | Complete | Require writing a script to utilize FPM to start with and later implement in pure python code. <br><br>20220204: We might change to rpmbuild directly without using FPM. See comments. | FPM usages
The code will also add dependencies to the RPM package so that things like JDK and additional libs for plugins can be installed and pulled separately | 5 | Complete | Need to study on RPM dependency setups | RPM Build Dependencies and the locations of each dependent artifact
The code will move the RPM package from the temp location to dist folder | 2 | Complete |   |

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add dashboards rpm assemble support in opensearch-build repo
 
### Issues Resolved
#1545
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
@spotrh could you help check the dashboards spec file again.
Thanks.

File: src/assemble_workflow/bundle_rpm.py [hunk 1]
```diff
@@ -89,7 +89,10 @@ def build(self, name: str, dest: str, archive_path: str, build_cls: BuildManifes
         os.environ.pop('OPENSEARCH_PATH_CONF', None)
 
         # Restore config file and core folder to original location
-        shutil.move(f"{min_bin_env_path}.backup", min_bin_env_path)
+        if os.path.exists(f"{min_bin_env_path}.backup"):
+            logging.info(f"Restore {min_bin_env_path}.backup to {min_bin_env_path}")
+            shutil.move(f"{min_bin_env_path}.backup", min_bin_env_path)
```
Comment (dblock) [inline src/assemble_workflow/bundle_rpm.py:94]: This looks hacky, why is any of this necessary? Why is it undoing something done in extract?

**Comment by peterzhuamazon** [comment]
> Do you plan to fix the "missing starting space in comment"?

That is straight up copy directly from OpenSearch core and it is expected.

**Comment by dblock** [comment]
> > Do you plan to fix the "missing starting space in comment"?
> 
> That is straight up copy directly from OpenSearch core and it is expected.

Why not fix it everywhere so that the YML look cleaner?

File: src/assemble_workflow/bundle_rpm.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/assemble_workflow/bundle_rpm.py:94]: This was in for the opensearch only change.
The reason being there is a hardcoded `source /etc/sysconfig/opensearch` line specifically in rpm version of core.
This caused the failure during installing of plugin zips.

In extract block we comment out the line and back up the original file.
Here we just move the backup back.

The change for dashboards is just to check if the file exist, if not then ignore this step.

Thanks.

**Comment by spotrh** [comment]
> 

Looks good to me without the /tmp permission item in the scriptlet.

**Comment by peterzhuamazon** [comment]
> > > Do you plan to fix the "missing starting space in comment"?
> > 
> > 
> > That is straight up copy directly from OpenSearch core and it is expected.
> 
> Why not fix it everywhere so that the YML look cleaner? Can be another PR.
> 
> Why are we copy-pasting an entire file? Doesn't this duplicate the original source of the documentation for it?

OpenSearch core has an empty file of opensearch_dashboards.yml.
As for bundle with plugins we are using the one in config folder of this repo.
I just copy paste the comments here so if this change is not seems to be necessary, I can revert.

This just seems like a good way for user to have a template instead of finding the settings themselves.

**Comment by peterzhuamazon** [comment]
> > 
> 
> Looks good to me without the /tmp permission item in the scriptlet.

Thanks @spotrh I have removed the /tmp line now :)

File: src/assemble_workflow/bundle_rpm.py (hunk 1 above)
Comment (dblock) [reply src/assemble_workflow/bundle_rpm.py:94]: Why isn't that line removed in the OpenSearch repo in the first place, why is this happening here?

I'll let it slide, but this is a hack on top of a hack.

**Review by dblock — APPROVED** [review]

**Review by tianleh — APPROVED** [review]

File: src/assemble_workflow/bundle_rpm.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/assemble_workflow/bundle_rpm.py:94]: Improvement issue: https://github.com/opensearch-project/opensearch-build/issues/1864

**Pull request merged**